driver-core/src/main/java/com/datastax/driver/core/Statement.java

```diff
@@ -252,4 +261,54 @@ public Statement setFetchSize(int fetchSize) {
     public int getFetchSize() {
         return fetchSize;
     }
-}
+
+    /**
+     * Defining the paging state for this statement.
+     *
+     * <p>
+     * Defining a paging state allows you to get the results of a statement
+     * from where you left it before (you get the next page of results from Cassandra,
+     * according to the @code{fetchsize}).
+     * Hence you should only use this method to set a paging state you had
+     * previously saved from the result of an execution ({@link com.datastax.driver.core.ExecutionInfo#getPagingState}).
+     * <p>
+     * Note : the paging state of a result is not supposed to be modified, it must only be saved to
+     * be re-used later, changing the content of the paging state will cause the hash MD5
+     * verification to fail and the paging state will be unusable.
+     *
+     * @param userPagingState state to set for the next query, if setFetchSize() is specified
+     *                        you will get the next page of results of size fetchSize. If not,
+     *                        the default fetchSize will be used as for a normal new Statement.
+     * @return the original statement set with the new paging state value.
+     */
+    public Statement setPagingState(PagingState userPagingState) {
+        if (this instanceof BatchStatement)
+            throw new UnsupportedOperationException("You cannot use the Cassandra pager on Batch statements");
+        else {
+            if (userPagingState.matches(this)) {
+                this.userPagingState = userPagingState.getRawState();
+            } else {
+                throw new InvalidPagingStateException("PagingState verification failed, this means that : either "
```

Comment:
Since this exception will always have the same message, you can make it constructor parameterless and inline the message in it:

    public InvalidPagingStateException() { super("PagingState verification failed..."); }